File: plugins/zego-assistant/skills/integrate-zego-server-api/examples/signature.py
"""ZEGO Server API Signature Generation"""

import secrets
import hashlib
import time


# Signature = md5(AppId + SignatureNonce + ServerSecret + Timestamp)
def generate_signature(app_id: int, signature_nonce: str, server_secret: str, timestamp: int) -> str:
    """Generate ZEGO API signature"""
    data = f"{app_id}{signature_nonce}{server_secret}{timestamp}"
    return hashlib.md5(data.encode("utf-8")).hexdigest()


def generate_signature_nonce() -> str:
    """Generate a 16-character hex string (8 bytes)"""
    return secrets.token_hex(8)


def main():
    # Example usage
    app_id = 12345
    server_secret = "your_server_secret"
    timestamp = int(time.time())
    nonce = generate_signature_nonce()
    signature = generate_signature(app_id, nonce, server_secret, timestamp)
    print(f"Signature: {signature}")


if __name__ == "__main__":
    main()

File: plugins/zego-assistant/skills/integrate-zego-server-api/examples/signature.ts
/**
 * ZEGO Server API Signature Generation
 */

import crypto from 'crypto';

// Signature = md5(AppId + SignatureNonce + ServerSecret + Timestamp)
export function generateSignature(
  appId: number,
  signatureNonce: string,
  serverSecret: string,
  timestamp: number
): string {
  const hash = crypto.createHash('md5');
  const data = `${appId}${signatureNonce}${serverSecret}${timestamp}`;
  hash.update(data);
  return hash.digest('hex');
}

// Generate a 16-character hex string (8 bytes)
export function generateSignatureNonce(): string {
  return crypto.randomBytes(8).toString('hex');
}

// Example usage
/*
const appId = 12345;
const serverSecret = 'your_server_secret';
const timestamp = Math.floor(Date.now() / 1000);
const nonce = generateSignatureNonce();
const signature = generateSignature(appId, nonce, serverSecret, timestamp);
console.log('Signature:', signature);
*/
